src/main/java/com/io/gittracker/controllers/MainViewPresenter.java

```diff
@@ -97,112 +140,89 @@ void createDefaultContent() {
+        VBox vBox = new VBox(panes);
+        vBox.getStyleClass().add("group-view-vbox");
+        vBox.prefHeight(0);
+        group.getRepositoriesProperty().addListener(createGithubRepositoryListChangeListener(vBox));
+        return vBox;
     }
 
-    @FXML
-    private TilePane repoBox;
-
-    public void createTilesFromList() {
-        clearTileList();
-        if (this.currentWorkspace == null) {
-            return;
-        }
-        this.currentWorkspace.getGroups().stream()
-                .flatMap(group -> group.getRepositories().stream())
-                .forEach(repo -> {
-                    if (repo != null) {
-                        repoBox.getChildren().add(createTile(repo));
-                    }
-                });
+    /** Listen to changes and create a view for new repositories  */
+    private ListChangeListener<GithubRepository> createGithubRepositoryListChangeListener(VBox vBox) {
+        return (ListChangeListener.Change<? extends GithubRepository> change) -> {
+            while (change.next()) {
+                if (change.wasAdded()) {
+                    List<? extends GithubRepository> addedRepositories = change.getAddedSubList();
+                    addedRepositories.forEach(repo -> vBox.getChildren().add(createRepositoryPane(repo)));
+                }
+            }
+        };
     }
 
-    private void clearTileList() {
-        this.repoBox.getChildren().clear();
+    /** Listen to changes and create a view for new groups  */
+    private ListChangeListener<Group> createGroupListChangeListener(VBox vBox) {
+        return (ListChangeListener.Change<? extends Group> change) -> {
+            while (change.next()) {
+                if (change.wasAdded()) {
+                    List<? extends Group> addedGroups = change.getAddedSubList();
+                    addedGroups.forEach(repo -> vBox.getChildren().add(createGroupView(repo)));
+                }
+            }
+        };
     }
 
-    private VBox createTile(GithubRepository repo) {
-        VBox tile = new VBox();
-        tile.getStyleClass().add("repoTile");
-        HBox upperRow = new HBox();
-        HBox lowerRow = new HBox();
-        Label repoName = new Label(repo.getName());
-        repoName.getStyleClass().add("clickable");
-        repoName.setOnMouseClicked(event -> {
-            System.out.println(repo.getHtmlUrl().toString());
-            this.hostServices.showDocument(repo.getHtmlUrl().toString());
-        });
-        Label lastCommit = new Label();
-        var latest_pr_optional = repo.getLatestPullRequest();
-        latest_pr_optional.ifPresentOrElse(
-                pullRequest -> {
-                    lastCommit.setText(String.format(pullRequest.getUpdatedAtDate() + " | " + pullRequest.getTitle()));
-                },
-                () -> {
-                    lastCommit.setText("Last PR unknown");
-                });
-        Button deleteButton = new Button();
-        deleteButton.setText("X");
```

Comment:
Delete button disappeared

Reply:
You're right, I'll add it back later XDD